OJS.Workers.ExecutionStrategies/BaseCodeExecutionStrategy.cs

```diff
@@ -38,6 +40,18 @@ protected virtual string SaveCodeToTempFile<TINput>(IExecutionContext<TINput> ex
                 ? FileHelpers.SaveStringToTempFile(this.WorkingDirectory, executionContext.Code)
                 : FileHelpers.SaveByteArrayToTempFile(this.WorkingDirectory, executionContext.FileContent);
 
+        protected void SaveZipSubmission(byte[] submissionContent, string directory)
+        {
+            var submissionFilePath = FileHelpers.SaveByteArrayToFileInDirectory(
+                directory,
+                ZippedSubmissionName,
+                submissionContent);
+
+            FileHelpers.RemoveFilesFromZip(submissionFilePath, RemoveMacFolderPattern);
+            FileHelpers.UnzipFile(submissionFilePath, directory);
+            FileHelpers.DeleteFiles(submissionFilePath);
```

Comment:
This gives the impression that we remove all files from the zip, we unzip all files (which are not removed/deleted) from the zip and then delete those files, which seems faulted. Is this an inaccuracy in the naming, or am I failing to get the logical flow.